api/v2/controllers/channels/channels_controller.py

```diff
@@ -1,10 +1,73 @@
+import os
+import uuid
+
 from flask import make_response, jsonify
-from flask_restful import Resource
+from flask_restful import Resource, request
+from googleapiclient.errors import HttpError
+
+from api.v2.helpers.credentials import check_bouquet_credentials
 from api.v2.helpers.channels.channels_helper import query_all_channels
+from api.v2.models.channels.channels_model import Channels as channels_model
+from api.v2.helpers.bouquets.bouquets_helper import get_bouquet
+from api.v2.utilities.validators import validate_calendar_info
 
 
 class Channels(Resource):
     def get(self):
         channels = query_all_channels()
+        return make_response(jsonify({'channels': channels}), 200)
+
+    def post(self):
+        # Enpoint to add channels to a bouquet
+        try:
+            calendar = request.get_json()
+
+            # Function call to validate channel data
+            is_valid = validate_calendar_info(calendar)
+            if is_valid.errors:
+                return {'error': is_valid.errors}, 400
+
+            # Function call to retrieve a bouquet
+            bouquet = get_bouquet(calendar['bouquet_id'])
+            if not bouquet:
+                return {'error': f"bouquet with id={calendar['bouquet_id']} doesn't exist"}, 404
+
+            if bouquet['should_refresh']:
+                return {'error': "can't manually add calendar to this bouquet"}, 403
+
+            bouquet_credentials = check_bouquet_credentials(bouquet)
+            service = bouquet_credentials['service']
+
+            channel = channels_model.query.filter_by(
+                                   calendar_id=calendar['calendar_id']).first()
+
+            if channel:
+                return {'error': 'calendar already in the bouquet'}, 409
+
+            channel = service.events().watch(
+                    calendarId=calendar['calendar_id'], body={
+                        'id': str(uuid.uuid1()),
+                        'type': 'web_hook',
+                        'address': os.getenv('NOTIFICATION_URL')
+                    }).execute()
+
+            if channel:
+                # save calendar to the bouquet
+                add_channel = channels_model.save_channel(
```

Comment:
This is a bad variable name. `add_channel` is a verb and as per our best practice, would best denote a function name.